README.release

```diff
@@ -0,0 +1,79 @@
+* Ensure version is incremented in:
+  configure.ac
+  docsrc/conf.py
```

Comment:
The version of this document for cyrus-imapd ([here](https://www.cyrusimap.org/imap/developer/releasing.html)) goes into more detail about the docsrc/conf.py juggling that's required across the repositories.  I don't know that we want to duplicate that here, but maybe we could reference it so some future maintainer can find it.

For that matter, maybe the right thing to do is document the juggling in the file, so someone grepping for version numbers finds the instructions alongside the place they're changing.  In fact I'm just gonna open an issue to remind myself to do that... https://github.com/cyrusimap/cyrus-imapd/issues/3909